Task3/main.py

```diff
@@ -0,0 +1,70 @@
+# ЗАДАЧА-1
+# Написать свой декоратор который будет проверять остаток от деления числа 100 на результат работы функции ниже.
+# Если остаток от деления = 0, вывести сообщение "We are OK!», иначе «Bad news guys, we got {}» остаток от деления.
+import functools
+
+
+def decorator(func):
+    @functools.wraps(func)
+    def wrapper(arg):
+        result = func(arg)
+        if result != 0:
+            print(f"result_decor = {100 % result}")
+            return result
+        else:
+            print("Результат функции decorator: 0")
+            return result
+    return wrapper
+
+# ЗАДАЧА-2
+# Написать декоратор который будет выполнять предпроверку типа аргумента который передается в вашу функцию.
+# Если это int, тогда выполнить функцию и вывести результат, если это str(),
+# тогда зарейзить ошибку ValueError (raise ValueError(“string type is not supported”))
+
+def chek_type_arg(func):
+    @functools.wraps(func)
+    def wrapper(arg):
+        if type(arg) is int:
+            result = func(arg)
+            print(f"Результат функции chek_type_arg: {result}")
+            #return result
+        else:
+            raise ValueError("string type is not supported")
```

Comment:
It works if we assume that only strings and integers are passed to the function. I would add "if isinstance(arg, str):" check to be sure that no other types of data are put as function args.